README.md

```diff
@@ -88,6 +88,22 @@ To export to an alternative file, modify the `args`:
       args: ["--frozen", "--output-file=requirements-custom.txt"]
 ```
 
+To synchronize your dependencies upon branch checkout or pull:
+
+```yaml
+default_install_hook_types:
+  - pre-commit
+  - post-checkout
+  - post-merge
+default_stages:
+  - pre-commit
+- repo: https://github.com/astral-sh/uv-pre-commit
+  # uv version.
+  rev: 0.4.29
+  hooks:
+    - id: uv-sync
+```
```

Comment:
Might be good to mention that if you have a uv workspace you'll want to add `--all-packages` to your args.